### ISSUE TITLE
`yarn unlink package` doesn't restore node_modules/package

### ISSUE THREAD

**Issue by whitecolor** [yarnpkg/yarn#1957]
After `yarn link package` yarn creates symlink node_module/package, but after `yarn unlink package` it just removes `node_module/package` and not places there hard installed version. 

Probably expected that it should find in cache or fetch a correct version from registry and replace removed symlink with it.

**Comment by whitecolor** [comment]
Seem to be the same issue https://github.com/yarnpkg/yarn/issues/937

**Comment by caesarsol** [comment]
+1, same issue here!

**Comment by andybluntish** [comment]
From the docs (https://yarnpkg.com/en/docs/cli/unlink#toc-yarn-unlink-package) this seems to be expected behaviour. However, as noted in #937 running `yarn` does not add the package back to node_modules.

**Comment by caesarsol** [comment]
Yes, it seems `yarn check` detects the package is missing, but `yarn install` does not restore it...

I have to run `yarn install --force` to make it right.

(As a possible improvement, I think `unlink` could reinstall the original package by itself)

**Comment by whitecolor** [comment]
`unlink` should reinstall.

**Comment by danielepiccone** [comment]
it works with a force install, but it surprises me why it doesn't with a normal `yarn install`

**Comment by stephencroberts** [comment]
Running `yarn install --check-files` seems to restore the package in v1.4.0.

**Comment by remino** [comment]
This has been going on since 2016? I'm having the same issue with yarn 1.12.3.

**Comment by arcanis** [comment]
I don't think `yarn unlink package` is meant to trigger an install (or at least, I'd feel uncomfortable changing this behavior anywhere else than in a major release). In the case where multiple packages are linked / unlinked (mostly scripts), running an new install every time would cause a perf hit.

That said, `yarn install` not noticing that a link disappeared is a bug. I don't have time right now to work on this, but #6931 seems an acceptable workaround for the time being (except that I'd recommend to use `--force` rather than `--check-files`, since the later has long-term effects).

**Comment by isakaweski** [comment]
Still not working on 2020.
Boooooring.

**Comment by dalmo3** [comment]
None of the suggestions above work for yarn v2 or v3 - most of those flags don't even exist. The only option is the going nuclear: 
```
rm -rf node_modules 
yarn
```
